tests/integration/test_pip.py

```diff
@@ -89,3 +91,92 @@ def test_pip_packages(
+        # in built image
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/pip-e2e-test.git",
+                ref="43ff06fdc751ebbbd57f9fe22dc9cc6b5d49a508",
+                packages=(
+                    {
+                        "type": "pip",
+                        "requirements_files": ["requirements.txt"],
+                        "requirements_build_files": [
+                            "requirements-git.txt",
+                            "requirements-https.txt",
+                        ],
+                    },
+                ),
+                check_vendor_checksums=False,
+                expected_exit_code=0,
+                expected_output="All dependencies fetched successfully",
+            ),
+            ["python3", "-m", "pip", "freeze"],
+            id="pip_e2e_test",
+        ),
+    ],
+)
+def test_e2e_pip(
+    test_params: utils.TestParameters,
+    check_cmd: List[str],
+    cachi2_image: utils.ContainerImage,
+    tmpdir: Path,
+    test_data_dir: Path,
+    request: pytest.FixtureRequest,
+):
+    """
+    End to end test for gomod.
+
+    :param test_params: Test case arguments
+    :param tmpdir: Temp directory for pytest
+    """
+    test_case = request.node.callspec.id
+
+    source_folder = utils.clone_repository(
+        test_params.repo, test_params.ref, f"{test_case}-source", tmpdir
+    )
+
+    output_folder = utils.fetch_deps_and_check_output(
+        tmpdir, test_case, test_params, source_folder, test_data_dir, cachi2_image
+    )
+
+    log.info("Create cachi2.env file")
+    env_vars_file = os.path.join(tmpdir, "cachi2.env")
+    cmd = [
+        "generate-env",
+        output_folder,
+        "--output",
+        env_vars_file,
+        "--for-output-dir",
+        os.path.join("/tmp", f"{test_case}-output"),
+    ]
+    (output, exit_code) = cachi2_image.run_cmd_on_image(cmd, tmpdir)
+    assert exit_code == 0, f"Env var file creation failed. output-cmd: {output}"
+
+    log.info("Inject project files")
+    cmd = [
+        "inject-files",
+        output_folder,
+        "--for-output-dir",
+        os.path.join("/tmp", f"{test_case}-output"),
+    ]
+    (output, exit_code) = cachi2_image.run_cmd_on_image(cmd, tmpdir)
+    assert exit_code == 0, f"Injecting project files failed. output-cmd: {output}"
+
+    log.info("Build container image with all prerequisites retrieved in previous steps")
+    container_folder = os.path.join(test_data_dir, test_case, "container")
+
+    with utils.build_image(
+        tmpdir, os.path.join(container_folder, "Containerfile"), test_case
+    ) as test_image:
+        log.info(f"Run command {check_cmd} on built image {test_image.repository}")
+        (output, exit_code) = test_image.run_cmd_on_image(check_cmd, tmpdir)
+        assert exit_code == 0, f"{check_cmd} command failed, Output: {output}"
+        for package in ["six", "ruamel", "operator-manifest", "dockerfile-parse"]:
```

Comment:
I think checking that the packages are installed properly is important, but could be done with just the CMD

One idea would be to execute [packages.py](https://github.com/cachito-testing/pip-e2e-test/blob/main/packages.py) (but update the script to fail if the imports fail rather than silently ignoring that)